config/checkstyle-checks.xml

```diff
@@ -484,6 +484,18 @@
                value="'@noinspection' Javadoc tags should be accompanied by a
                      '@noinspectionreason' tag, explaining why we suppressed inspection."/>
     </module>
+    <module name="MatchXpath">
```

Comment:
Couldn't this be created as an ArchUnit?

Reply:
good point.
@stoyanK7 , please look at https://github.com/checkstyle/checkstyle/blob/master/src/test/java/com/puppycrawl/tools/checkstyle/internal/ArchUnitTest.java

it might a way more reliable way to do this, as it is executing in runtime and all classes and fields are available by reflection.

Reply:
There are 3 suggestions to do the same thing. `MatchXpath`, `testAllCheckstyleChecksHaveMessage` and `ArchUnitTest`. Which one shall we go with?

Reply:
We can do duplicates. I am fine to ignore ArchUnit for `testAllCheckstyleChecksHaveMessage`. `testAllCheckstyleChecksHaveMessage` will ensure we get all checks.

Reply:
Done `testAllCheckstyleChecksHaveMessage` is extended. Here is a failure message when I remove an annotation
```
MSG_KEY in OneStatementPerLineCheck should have @MessageKey annotation
expected not to be empty

```